p2p/protocol/identify/obsaddr.go

```diff
@@ -435,6 +441,20 @@ func shouldRecordObservation(host addrsProvider, network listenAddrsProvider, co
 	return true
 }
 
+// isNAT64IPv4ConvertedIPv6Addr returns whether addr is an IPv6 address that begins with
+// the well known prefix "64:ff9b" used for NAT64 Translation
+// see RFC 6052
+func isNAT64IPv4ConvertedIPv6Addr(addr ma.Multiaddr) bool {
+	ip, err := addr.ValueForProtocol(ma.P_IP6)
+	if err != nil {
+		return false
+	}
+	if strings.HasPrefix(ip, "64:ff9b::") {
```

Comment:
Is string matching the best way to do this? Would be nicer to match on the underlying bytes (and probably also causes fewer allocs?).

Reply:
Ok, let's keep it for now, but we should keep this is mind for our multiaddrs API redesign. It's absurd that we return IPs as strings.